### PR TITLE
More documentation updates

### PR DESCRIPTION
**Pull request by jamie-lemon**
- Removes `stories-api.rst` and supplants info into the header of `xml-class.rst`
- Removes any references to `stories-api.rst`
- Tidies up URL links in `tutorial.rst`
- Fixes footnotes in `story-class.rst` to match how we do footnotes elsewhere
- Changes URL link to mupdf.com to https
- A couple of cosmetic issues

### CODE REVIEW

**Comment by jamie-lemon** [comment]
Decided that going through all the docs and moving the external URLs to the page bottom was just too painful to do now. but will add to the general docs TODO list.

File: docs/tutorial.rst [hunk 1]
```diff
@@ -64,9 +64,9 @@ creator        creating application
 subject        subject
 ============== =================================
 
-.. note:: Apart from these standard metadata, **PDF documents** starting from PDF version 1.4 may also contain so-called *"metadata streams"* (see also :data:`stream`). Information in such streams is coded in XML. PyMuPDF deliberately contains no XML components, so we do not directly support access to information contained therein. But you can extract the stream as a whole, inspect or modify it using a package like `lxml <https://pypi.org/project/lxml/>`_ and then store the result back into the PDF. If you want, you can also delete these data altogether.
+.. note:: Apart from these standard metadata, **PDF documents** starting from PDF version 1.4 may also contain so-called *"metadata streams"* (see also :data:`stream`). Information in such streams is coded in XML. PyMuPDF deliberately contains no XML components, so we do not directly support access to information contained therein. But you can extract the stream as a whole, inspect or modify it using a package like `lxml`_ and then store the result back into the PDF. If you want, you can also delete these data altogether.
```
Comment (JorjMcKie) [inline docs/tutorial.rst:67]: We now do have XML components, so we should adapt to "... PyMuPDF deliberately contains no XML components for this purpose, ..."

**Review by JorjMcKie — CHANGES_REQUESTED** [review]
We should adapt a formulation in `tutorial.rst`, when we say "... PyMupdf contains no XML component, ...".
This obviously no longer true, so I suggest to now say this: "... PyMupdf contains no XML component for this purpose, ...".

**Comment by jamie-lemon** [comment]
Okay - updated ...

**Review by JorjMcKie — APPROVED** [review]

**Pull request merged**